src/libYARP_OS/src/Timer.cpp

```diff
@@ -0,0 +1,48 @@
+#include <yarp/os/Timer.h>
+#include <yarp/os/Time.h>
+using namespace yarp::os;
+
+Timer::Timer(const double rate, TimerCallback callback, Mutex* mutex, const bool inputOneshot) :
+    RateThread(rate),
+    m_callback(callback),
+    m_oneshot(inputOneshot),
+    m_mutex(mutex)
+{}
+
+
+bool Timer::threadInit()
+{
+    m_startStamp = yarp::os::Time::now();
+    return true;
+}
+
+void Timer::run()
+{
+    if(m_runCount == 0)
+    {
+        m_runCount++;
+        return;
+    }
```

Comment:
Please add a comment to this "if"...
I understand the reason for it, but I'm pretty sure that in the future someone will see this and will remove it without understanding the reason 😉